testsuite/tests/typing-warnings/unused_types.ml

```diff
@@ -671,3 +671,21 @@ Warning 34 [unused-type-declaration]: unused type "unused2".
 
 val f : 'a -> 'a = <fun>
 |}]
+
+let f (x1 : int as 'unused1) (x2 : int as 'unused2)
+      (x3 : int as 'used1) (x4 : 'used1) (x5 : 'used2) = (x1, x2, x3, x4, x5)
+
+[%%expect{|
+Line 1, characters 19-27:
+1 | let f (x1 : int as 'unused1) (x2 : int as 'unused2)
+                       ^^^^^^^^
+Warning 34 [unused-type-declaration]: unused type variable "'unused1".
```

Comment:
Minor remark: the message is fine, if we were to extend the warning to all unused type variables in expressions.
Currently, it might create the expectation that the same warning is raised for
```ocaml
let f (x:'a) (y:'b) = ()
```
which is not the case. Thus I wonder, if we might want to specialize the message to
```ocaml
Warning 34 [unused-type-declaration]: unused type alias `'unused1`.
```